ToDo.Core/ViewModels/AddViewModel.cs

```diff
@@ -18,18 +22,110 @@ public class AddViewModel : MvxViewModel
         private readonly IToDoService _todoService;
         private readonly IMvxNavigationService _navigationService;
         private readonly IUserDialogs _userDialogService;
+        private readonly IComputerVisionService _computerVisionService;
+        private readonly IMediaService _mediaService;
 
         public Models.ToDo ToDoItem { get; set; }
         public ICommand AddTodoItemCommand { get; }
+        public ICommand TakePhotoCommand { get; }
 
-        public AddViewModel(IToDoService todoService, IMvxNavigationService navigationService, IUserDialogs userDialogService)
+        private Stream photoStream;
+
+        public AddViewModel(IToDoService todoService, IMvxNavigationService navigationService, IUserDialogs userDialogService,
+            IComputerVisionService computerVisionService, IMediaService mediaService)
```

Comment:
You see now how fast the list of dependencies of a viewmodel adds up.
There are ways, you can get around this though: Using a Mediator pattern:
https://lostechies.com/jimmybogard/2015/05/05/cqrs-with-mediatr-and-automapper/ 
We are thinking about using this approach for our app as well ;-)